Code/Framework/AzCore/Platform/Common/UnixLike/AzCore/IO/SystemFile_UnixLike.cpp

```diff
@@ -223,6 +224,92 @@ namespace PosixInternal
     {
         return dup2(fileDescriptorSource, fileDescriptorDestination);
     }
+
+    int Pipe(int(&pipeFileDescriptors)[2], int, OpenFlags pipeFlags)
+    {
+        return pipe2(pipeFileDescriptors, static_cast<int>(pipeFlags));
+    }
 } // namespace AZ::IO::PosixInternal
 
+namespace AZ::IO
+{
+    // FileDescriptorCapturer UnixLike Impl
+    void FileDescriptorCapturer::Start(int pipeSize)
+    {
+        if (m_redirectToPipe)
+        {
+            // Capturer is already in progress
+            return;
+        }
+        if (m_sourceDescriptor == -1)
+        {
+            // Source file descriptor isn't set.
+            return;
+        }
+        int pipeCreated = PosixInternal::Pipe(m_pipe, pipeSize, PosixInternal::OpenFlags::NonBlock);
+
+        if (pipeCreated == -1)
+        {
+            return;
+        }
+
+        // Duplicate the original source descriptor to restore in Stop
+        m_dupSourceDescriptor = PosixInternal::Dup(m_sourceDescriptor);
+
+        // Duplicate the write end of the pipe onto the original source descriptor
+        // This causes the writes to the source descriptor to redirect to the pipe
+        if (PosixInternal::Dup2(m_pipe[WriteEnd], m_sourceDescriptor) == -1)
+        {
+            // Failed to redirect the source descriptor to the pipe
+            PosixInternal::Close(m_dupSourceDescriptor);
+            PosixInternal::Close(m_pipe[WriteEnd]);
+            PosixInternal::Close(m_pipe[ReadEnd]);
+            // Reset pipe descriptor to -1
+            m_pipe[WriteEnd] = -1;
+            m_pipe[ReadEnd] = -1;
+            m_dupSourceDescriptor = -1;
+            return;
+        }
+
+        m_redirectToPipe = true;
+    }
+
+    void FileDescriptorCapturer::Flush(const OutputRedirectVisitor& redirectCallback)
+    {
+        constexpr int PipeBufferSize = DefaultPipeSize;
+        AZStd::array<AZStd::byte, PipeBufferSize> capturedBytes;
+
+        int bytesRead{};
+        do
+        {
+            // Pump the read end of the pipe until it is empty
+            // and invoke the visitor for each call
+            bytesRead = PosixInternal::Read(m_pipe[ReadEnd],
+                AZStd::ranges::data(capturedBytes), static_cast<int>(AZStd::ranges::size(capturedBytes)));
+            if (bytesRead > 0)
+            {
+                if (redirectCallback)
```

Comment:
nit: can redirectCallback go away during this loop? should we just AZ_Assert(redirectCallback) at the beginning of the function instead before we enter the loop?